calicoctl/commands/edit.go

```diff
@@ -255,3 +263,21 @@ func addComment(b []byte, err error) []byte {
 	nb.Write(b)
 	return nb.Bytes()
 }
+
+// ValidateIDFieldsSame is used to validate that the Resource Objects
+// supplied in one slice reference the same objects in the other slice
+// by comparing the ID fields.
+func ValidateIDFieldsSame(original, changed []unversioned.ResourceObject) error {
+	if len(original) != len(changed) {
+		return errors.New("Number of resources after editing does not match the number of resources provided. Cannot add or remove resources during edit.")
+	}
+
+	// Assume that that the resources are kept in the same order
+	for i, _ := range original {
+		if original[i].String() != changed[i].String() {
+			return fmt.Errorf("Cannot change metadata fields: %s should match %s", changed[i].String(), original[i].String())
```

Comment:
not quite true - you are allowed to change the labels which are a metadata field.